firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestoreSettings.java

```diff
@@ -124,6 +130,25 @@ public Builder setTimestampsInSnapshotsEnabled(boolean value) {
       return this;
     }
 
+    /**
+     * Sets the cache size threshold above which the SDK will attempt to collect least-recently-used
+     * documents. The size is not a guarantee that the cache will stay below that size, only that if
+     * the cache exceeds the given size, cleanup will be attempted.
+     *
+     * @return A settings object on which the cache size is configured as specified by the given
+     *     {@code value}.
+     */
+    @NonNull
```

Comment:
I don't have strong feelings.  I'd maybe throw "an *approximate* cache size" into the first sentence.

Can we also mention the default size?

Reply:
I have reworded it a little and included the default. However:

It is not a size of the document cache, it's the size of all on-disk data.

It's not an approximate cache size for on-disk data. It's the exact size that the files are taking up. It will be approximate when we later make it possible to use LRU for in-memory persistence, but I expect that will be another change of some sort for the public api. In particular we will have to update the default, which will mean a change to this comment.

I also don't think the word temporarily is appropriate, since it is possible to outrun the garbage collector and never get under the threshold.

Reply:
While I don't disagree with any of that, I'd prioritize conveying the intention of this API to the developer (so they know how to use it) over conveying the exact implementation details (which are subject to change anyway).  At the end of the day, this is an API to help developers manage their on-disk cache size.  It's worth conveying that it's not a _hard_ limit. It's an _approximate_ limit.  But I don't think it's worth conveying exactly how we try to enforce the limit, in what ways it's only approximate, or edge cases where we don't successfully prevent their cache from growing unbounded, etc.

Reply:
It's not a limit at all, really. It's a threshold: above the threshold, we try to clean up. Under, we don't. It is I guess a limit on how large the cache can get before we will try to clean up.

To that end, I wouldn't (and don't) use the word limit in the comment. 

I have updated the first sentence to include the word _approximate_. I'm not sure that really improves much, since the threshold is not the thing that is approximate. Maybe there's another wording that could better convey that? I've tried to be clear that the cache _size_ is not a guarantee.

I have also included some more text on the minimum value and the unlimited settings, which I realized was missing.

Reply:
This isn't critical to get perfect so feel free to keep what you have (and definitely feel free to remove "approximate" since the threshold really isn't approximate).

That said, if I were to describe this feature in a simple sentence it'd be "Sets a maximum cache size." or "Sets a limit on the cache size."  But to avoid support requests saying it's not working, I'd add the word "approximate" to convey that it won't be strictly enforced.  So that's where my feedback came from.  But what you have is fine. 😁  🚢 🚢 🚢